File: Atcoder_#349/javascript/b_commencement.js
function main(input) {
  function isGoodString(s) {
    let counts = {};
    for (let char of s) {
      counts[char] = (counts[char] || 0) + 1;
    }

    let occurrenceCounts = Object.values(counts);

    let frequencyCounts = {};
    for (let count of occurrenceCounts) {
      frequencyCounts[count] = (frequencyCounts[count] || 0) + 1;
    }

    return Object.values(frequencyCounts).every(count => count === 0 || count === 2);
  }

  const s = input.trim();
  const result = isGoodString(s) ? "Yes" : "No";
  
  console.log(result);
}
main(require("fs").readFileSync("/dev/stdin", "utf8"));

File: Atcoder_#349/c/b_commencement.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool isGoodString(const char *s) {
    int counts[26] = {0};

    for (int i = 0; s[i] != '\0'; ++i) {
        counts[s[i] - 'a']++;
    }

    int frequencyCounts[101] = {0};

    for (int i = 0; i < 26; ++i) {
        if (counts[i] > 0) {
            frequencyCounts[counts[i]]++;
        }
    }

    for (int i = 1; i <= 100; ++i) {
        if (frequencyCounts[i] != 0 && frequencyCounts[i] != 2) {
            return false;
        }
    }

    return true;
}

int main() {
    char s[101];

    scanf("%100s", s);

    if (isGoodString(s)) {
        printf("Yes\n");
    } else {
        printf("No\n");
    }

    return 0;
}
